module/zfs/dmu_send.c

```diff
@@ -1244,12 +1248,25 @@ dmu_send_obj(const char *pool, uint64_t tosnap, uint64_t fromsnap,
 			dsl_pool_rele(dp, FTAG);
 			return (err);
 		}
-		if (!dsl_dataset_is_before(ds, fromds, 0))
-			err = SET_ERROR(EXDEV);
+		if (!dsl_dataset_is_before(ds, fromds, 0)) {
+			err = SET_ERROR(ENODEV);
```

Comment:
Was this error code changed deliberately?

Reply:
no, fixed.